pkg/lib/dialogs.jsx

```diff
@@ -115,10 +117,26 @@ export const WithDialogs = ({ children }) => {
                               JSON.stringify(dialog));
             is_open.current = !!component;
             setDialog(component);
+            return new Promise((resolve, reject) => {
+                setResolvePromise(() => resolve);
+                setRejectPromise(() => reject);
+            });
         },
-        close: () => {
+        close: (args) => {
             is_open.current = false;
+            // backwards compatibility with non-async functions
```

Comment:
I don't understand this. `resolvePromise` is only null when no dialog is actually open, no? What would be a example of a legacy non-async function?

Reply:
I have no idea why, but all tests where breaking on calling `resolvePromise` which was `null`.

Reply:
Interesting. There might be multiple calls to `Dialogs.close` in a row. I think checking for a non-null resolvePromise before calling it is just fine, but the comment doesn't make sense to me. Maybe "Protect against calling Dialogs.close() when no dialog is open" would be better, but I don't think we even need to explain this.